### PR TITLE
Add `group` field to `google_network_connectivity_spoke`

### PR DESCRIPTION
**Pull request by septikus**
- **Add `group` field to Spoke resource.**

Fixes https://github.com/hashicorp/terraform-provider-google/issues/18297

```release-note:enhancement
networkconnectivity: added `group` field to `google_network_connectivity_spoke` resource
```

### CODE REVIEW

File: mmv1/products/networkconnectivity/Spoke.yaml [hunk 1]
```diff
@@ -122,6 +128,10 @@ properties:
     immutable: true
     resource: 'hub'
     imports: 'name'
+  - name: 'group'
```
Comment (zli82016) [inline mmv1/products/networkconnectivity/Spoke.yaml:131]: Is this field updatable? If yes, can you please add the update test for this field in the [file](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/services/networkconnectivity/resource_network_connectivity_spoke_test.go) ? 

This is the [doc](https://googlecloudplatform.github.io/magic-modules/test/test/#add-an-update-test) to add an update test.

File: mmv1/products/networkconnectivity/Spoke.yaml [hunk 2]
```diff
@@ -44,6 +44,12 @@ examples:
       network_name: 'net'
       hub_name: 'hub1'
       spoke_name: 'spoke1'
+  - name: 'network_connectivity_spoke_linked_vpc_network_group'
+    primary_resource_id: 'primary'
+    vars:
+      network_name: 'net'
```
Comment (zli82016) [inline mmv1/products/networkconnectivity/Spoke.yaml:50]:
```suggestion
      network_name: 'net-spoke'
```

File: mmv1/products/networkconnectivity/Spoke.yaml [hunk 3]
```diff
@@ -44,6 +44,12 @@ examples:
...
+      hub_name: 'hub1'
```
Comment (zli82016) [inline mmv1/products/networkconnectivity/Spoke.yaml:51]:
```suggestion
      hub_name: 'hub1-spoke'
```

File: mmv1/products/networkconnectivity/Spoke.yaml [hunk 4]
```diff
@@ -44,6 +44,12 @@ examples:
...
+      spoke_name: 'spoke1'
```
Comment (zli82016) [inline mmv1/products/networkconnectivity/Spoke.yaml:52]:
```suggestion
      spoke_name: 'group-spoke1'
```

Comment (zli82016) [reply mmv1/products/networkconnectivity/Spoke.yaml:52]: This change is to add the prefix `tf-test-` to the name automatically. The prefix will help to sweep the leaked resources.

**Review by zli82016 — CHANGES_REQUESTED** [review]

**Comment by zli82016** [comment]
Is there a Github issue for this PR?

**Comment by ghabian** [comment]
https://github.com/hashicorp/terraform-provider-google/issues/20362

**Comment by septikus** [comment]
> Is there a Github issue for this PR?

https://github.com/hashicorp/terraform-provider-google/issues/18297

File: mmv1/products/networkconnectivity/Spoke.yaml (hunk 1 above)
Comment (septikus) [reply mmv1/products/networkconnectivity/Spoke.yaml:131]: The field is currently not updateable. That will come at a later time.

Comment (zli82016) [reply mmv1/products/networkconnectivity/Spoke.yaml:131]: Can you add `immutable: true` to it for this case?

Comment (septikus) [reply mmv1/products/networkconnectivity/Spoke.yaml:131]: Done!

**Review by zli82016 — APPROVED** [review]
LGTM

**Pull request merged**